TestLeaderAndVoter is flaky

## Flaky Test

### Which jobs are failing

```
    --- FAIL: TestRuleTestSuite/TestLeaderAndVoter (20.62s)
        testutil.go:67: 
            	Error Trace:	/home/runner/work/pd/pd/pkg/utils/testutil/testutil.go:67
            	            				/home/runner/work/pd/pd/tests/server/api/rule_test.go:982
            	            				/home/runner/work/pd/pd/tests/testutil.go:363
            	            				/home/runner/work/pd/pd/tests/testutil.go:323
            	            				/home/runner/work/pd/pd/tests/server/api/rule_test.go:903
            	Error:      	Condition never satisfied
            	Test:       	TestRuleTestSuite/TestLeaderAndVoter
```

### CI link

https://github.com/tikv/pd/actions/runs/9968939441/job/27545022545

### Reason for failure (if possible)


### Anything else

--- a/tests/server/api/rule_test.go
+++ b/tests/server/api/rule_test.go
@@ -985,13 +985,7 @@ func (suite *ruleTestSuite) checkLeaderAndVoter(cluster *tests.TestCluster) {
 				tu.StatusOK(re), tu.ExtractJSON(re, &respBundle))
 			re.NoError(err)
 			re.Len(respBundle, 1)
-			if bundle[0].Rules[0].Role == placement.Leader {
-				return respBundle[0].Rules[0].Role == placement.Leader
-			}
-			if bundle[0].Rules[0].Role == placement.Voter {
-				return respBundle[0].Rules[0].Role == placement.Voter
-			}
-			return false
+			return compareBundle(respBundle[0], bundle[0])
 		})
 	}
 }

